### DISCUSSION TITLE
Add Quick/Snap search to song select

### DISCUSSION THREAD

**Discussion by lisekilis** [ppy/osu#24903] [Ideas]
# It would be nice to see Quick/Snap search (I don't know the actual name of this feature) beeing added as an option in song select.

# What Is have in mind.
When you press a letter on your keyboard while in file explorer or your steam library it will instantly jump to the first file/game that starts with that letter and if you keep typing it will just to that file/game name.
Example No.1
![image](https://github.com/ppy/osu/assets/101191226/53c850b0-fcd1-44dc-885b-d9feebd5dadc) > ![image](https://github.com/ppy/osu/assets/101191226/3ddc6a59-7448-42f9-8789-b13c58c6f794) > ![image](https://github.com/ppy/osu/assets/101191226/b702a591-a5de-49d5-a35e-d82ad5e4c3ba)
Example No.2
![image](https://github.com/ppy/osu/assets/101191226/53c850b0-fcd1-44dc-885b-d9feebd5dadc) >  ![image](https://github.com/ppy/osu/assets/101191226/3ddc6a59-7448-42f9-8789-b13c58c6f794) ![image](https://github.com/ppy/osu/assets/101191226/d9d406e2-faf8-4c44-a515-0d97ed854e88) ![image](https://github.com/ppy/osu/assets/101191226/32f57e0e-5f2d-4f04-8dbb-32b6280743a6) >![image](https://github.com/ppy/osu/assets/101191226/6102bba3-28cd-4540-9726-1cb840cd8a84)
# Exepected impementation in osu!lazer:
Quick search would be on if you disable automatic search select which would be on by default as it is the current behaviour.
![osu](https://github.com/ppy/osu/assets/101191226/84f4ff10-5282-4e5d-953b-daa472a06ac8)

Hopefully this feature makes it into osu! lazer some time.

**Comment by peppy** [comment]
Not sure about this one, it's a fundamental change to search implementation, and I think it would be a worse experience for the majority of searches (where results are not nearby in the carousel and would be very abrupt to jump between.

**Reply by lisekilis** [reply]
I always found it wierd that basically no games have this feature since I am really used to using it and I know a lot of people who also use this for quick and accurate searches. I do also see that the search system would have to undergo heavy change to be implemented and it's probbably not worth your time (at least in the near future since the road to ranked play)
Thanks for the response tho!
